fmt/src/formatter.rs

```diff
@@ -1461,14 +1461,84 @@ impl<'a, W: Write> Visitor for Formatter<'a, W> {
                 }
                 Type::Function { .. } => self.visit_source(*loc)?,
             },
+            Expression::Unit(_, expr, unit) => {
+                expr.visit(self)?;
+                let (loc, str) = match unit {
+                    Unit::Seconds(loc) => (loc, "seconds"),
+                    Unit::Minutes(loc) => (loc, "minutes"),
+                    Unit::Hours(loc) => (loc, "hours"),
+                    Unit::Days(loc) => (loc, "days"),
+                    Unit::Weeks(loc) => (loc, "weeks"),
+                    Unit::Wei(loc) => (loc, "wei"),
+                    Unit::Gwei(loc) => (loc, "gwei"),
+                    Unit::Ether(loc) => (loc, "ether"),
+                };
```

Comment:
is it worth it to convert this into two functions `loc` + AsRef<str>?

Reply:
good idea! doing it rn